nsqd/nsqd.go

```diff
@@ -218,14 +222,17 @@ func (n *NSQD) IsHealthy() bool {
 }
 
 func (n *NSQD) GetError() error {
-	n.errMtx.RLock()
-	defer n.errMtx.RUnlock()
-	return n.err
+	errValue := n.errValue.Load()
+	if errValue == nil {
```

Comment:
considering you can't ever `Store(nil)`, you could initialize errValue to a empty errStore, and then get rid of this check